src/formatDate.js

```diff
@@ -9,6 +9,46 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    dateMap[fromFormat[i]] = dateParts[i];
+  }
```

Comment:
Consider using array destructuring with the `forEach` method or a `for...of` loop for better readability when mapping date parts to their corresponding format. This would avoid the need for manual index management and make the code cleaner.